File: C++/TicTacToe/TicTacToe.cpp
#include <iostream>
using namespace std;

char board[9] = {'1','2','3','4','5','6','7','8','9'};
char x_mark = 'X', o_mark = 'O';
char current = x_mark;

void drawBoard () {
	system ("clear"); // use "cls" on Windows
	cout << " " << board[0] << " │ " << board[1] << " │ " << board[2] << "\n";
	cout << "───┼───┼───\n";
	cout << " " << board[3] << " │ " << board[4] << " │ " << board[5] << "\n";
	cout << "───┼───┼───\n";
	cout << " " << board[6] << " │ " << board[7] << " │ " << board[8] << "\n\n";
}

// Check all 8 winning combos
bool checkWin () {
	int wins[8][3] = {
		{0,1,2},{3,4,5},{6,7,8},
		{0,3,6},{1,4,7},{2,5,8},
		{0,4,8},{2,4,6}
	};

	for (auto &w : wins) {
		if (board[w[0]] == board[w[1]] && board[w[1]] == board[w[2]])
			return true;
	}

	return false;
}

bool isDraw () {
	for (char c : board)
		if (c != x_mark && c != o_mark)
			return false;

	return true;
}

int main () {
	while (true) {
		drawBoard ();
		cout << "Player " << current << ", enter a position (1-9): ";

		int pos;
		cin >> pos;

		if (pos < 1 || pos > 9)
			cout << "Invalid input!"
			continue;

		if (board[pos-1] == x_mark || board[pos-1] == o_mark)
			cout << "Spot taken! Choose another.";
			continue;

		board[pos-1] = current;

		if (checkWin ()) {
			drawBoard();
			cout << "Player " << current << " wins!\n";
			break;
		}

		if (isDraw ()) {
			drawBoard ();
			cout << "It's a draw!\n";
			break;
		}

		current = (current == x_mark) ? o_mark : x_mark;
	}
	return 0;
}

File: Python/TicTacToe/TicTacToe.py
board = [str(i) for i in range(1,10)]
x_mark = "X"
o_mark = "O"
current = x_mark

def draw_board():
	print(f" {board[0]} | {board[1]} | {board[2]}")
	print("---+---+---")
	print(f" {board[3]} | {board[4]} | {board[5]}")
	print("---+---+---")
	print(f" {board[6]} | {board[7]} | {board[8]}\n")

def check_win():
	wins = [
		(0,1,2),(3,4,5),(6,7,8),
		(0,3,6),(1,4,7),(2,5,8),
		(0,4,8),(2,4,6)
	]

	return any(board[a]==board[b]==board[c] for a,b,c in wins)

def is_draw():
	return all(s in (x_mark,o_mark) for s in board)

while True:
	draw_board()
	move = input(f"Player {current}, choose 1-9: ")

	if not move.isdigit() or not 1 <= int(move) <= 9:
		continue

	idx = int(move)-1

	if board[idx] in (x_mark,o_mark):
		continue

	board[idx] = current

	if check_win():
		draw_board()
		print(f"Player {current} wins!")
		break

	if is_draw():
		draw_board()
		print("It's a draw!")
		break

	current = o_mark if (current == x_mark) else x_mark